### PR TITLE
[FSSDK-10882] ProjectConfigManager SSR support

### PR DESCRIPTION
**Pull request by junaed-optimizely**
## Summary
This PR makes the `ProjectConfigManager` SSR compatible
## Test plan
- Existing mock has been updated to adjust with the config manager
- New testcases have been added for coverage
## Issues
- [FSSDK-10882](https://jira.sso.episerver.net/browse/FSSDK-10882)

### CODE REVIEW

File: lib/project_config/project_config_manager.ts [hunk 1]
```diff
@@ -79,6 +81,11 @@ export class ProjectConfigManagerImpl extends BaseService implements ProjectConf
       return;
     }
 
+    if(this.isSsr) {
```
Comment (raju-opti) [inline lib/project_config/project_config_manager.ts:84]: if isSsr is true no datafile is provided, but only a datafileManager is provided, then onRunning() will wait indefinitely cause we are dropping the datafile. We can move this condition before the previous if on line 78 and update the message inside for isSsr case

Comment (raju-opti) [reply lib/project_config/project_config_manager.ts:84]: Also can we add a test that if isSsr is provided without a datafile, onRunning() is rejected?

File: lib/optimizely/index.ts [hunk 2]
```diff
@@ -146,6 +146,7 @@ export default class Optimizely implements Client {
       this.updateOdpSettings();
     });
 
+    this.projectConfigManager.setSsr(config.isSsr)
```
Comment (raju-opti) [inline lib/optimizely/index.ts:149]: can we add a test that isSsr is passed to the projectConfigManager?

Please do this in a new lib/optimizely/index.spec.ts file instead of the old js test file. We will eventually get rid of the whole js test file.

File: lib/index.node.tests.js [hunk 3]
```diff
@@ -88,6 +89,22 @@ describe('optimizelyFactory', function() {
         assert.equal(optlyInstance.clientVersion, '5.3.4');
       });
 
+      it('should create an instance of optimizely with ssr flag, project config must be available', () => {
```
Comment (raju-opti) [inline lib/index.node.tests.js:92]: could you please add the test in a new index.node.spec.ts file instead? We want to get rid of all JS tests. We will only update existing tests in these old tests file and will add all new tests in new ts test files.

File: lib/project_config/project_config_manager.ts (hunk 1 above)
Comment (junaed-optimizely) [reply lib/project_config/project_config_manager.ts:84]: Good point. I have missed that edge case. Thanks for pointing it out..

File: lib/optimizely/index.ts (hunk 2 above)
Comment (junaed-optimizely) [reply lib/optimizely/index.ts:149]: Ok. I did not know about that change.

File: lib/index.node.tests.js (hunk 3 above)
Comment (junaed-optimizely) [reply lib/index.node.tests.js]: This won't be required anymore due to the new `index.spec.ts` coverage
